osc-server/src/test/java/org/osc/core/broker/service/BindSecurityGroupServiceTest.java

```diff
@@ -0,0 +1,574 @@
+        request.addServiceToBindTo(serviceToBindTo);
+        this.exception.expect(VmidcBrokerInvalidEntryException.class);
+        this.exception.expectMessage(field + " should not have an empty value.");
+
+        // Act
+        this.bindSecurityGroupService.dispatch(request);
+    }
+
+    @Test
+    public void testValidate_WhenServiceVirtualSystemIdNull_ThrowsInvalidEntryException() throws Exception {
+        // Arrange
+        String field = "Virtual System Id";
+        BindSecurityGroupRequest request = createRequest(VALID_ID, VALID_ID, null);
+        VirtualSystemPolicyBindingDto serviceToBindTo = createRequestWithService(null);
+        request.addServiceToBindTo(serviceToBindTo);
+        this.exception.expect(VmidcBrokerInvalidEntryException.class);
+        this.exception.expectMessage(field + " should not have an empty value.");
+
+        // Act
+        this.bindSecurityGroupService.dispatch(request);
+    }
+
+    @Test
+    public void testValidate_WhenServiceNameNull_ThrowsInvalidEntryException() throws Exception {
+        // Arrange
+        String field = "Service Name";
+        BindSecurityGroupRequest request = createRequest(VALID_ID, VALID_ID, null);
+        VirtualSystemPolicyBindingDto serviceToBindTo = createRequestWithService(VALID_ID);
+        serviceToBindTo.setName(null);
+        request.addServiceToBindTo(serviceToBindTo);
+        this.exception.expect(VmidcBrokerInvalidEntryException.class);
+        this.exception.expectMessage(field + " should not have an empty value.");
+
+        // Act
+        this.bindSecurityGroupService.dispatch(request);
+    }
+
+    @Test
+    public void testValidate_WithInValidSgId_ThrowsValidationException() throws Exception {
+        // Arrange
+        BindSecurityGroupRequest request = createRequest(VALID_ID, INVALID_ID, VALID_ID);
+        this.exception.expect(VmidcBrokerValidationException.class);
+        this.exception.expectMessage("Security Group with Id: " + request.getSecurityGroupId() + "  is not found.");
+
+        // Act
+        this.bindSecurityGroupService.dispatch(request);
+
+    }
+
+    @Test
+    public void testValidate_WhenSgIdsVcIdAndActualVcIdMisMatch_ThrowsValidationException() throws Exception {
+        // Arrange
+        BindSecurityGroupRequest request = createRequest(INVALID_ID, VALID_ID, VALID_ID);
+        this.exception.expect(VmidcBrokerValidationException.class);
+        this.exception
+                .expectMessage(String.format("The Security Group '%s' does not belong to the Parent Object with ID %d",
+                        DEFAULT_SG_NAME, INVALID_ID));
+
+        // Act
+        this.bindSecurityGroupService.dispatch(request);
+
+    }
+
+    @Test
+    public void testValidate_WhenSgIsMarkedForDeletion_ThrowsInvalidRequestException() throws Exception {
+        // Arrange
+        BindSecurityGroupRequest request = createRequest(VALID_ID_MARK_FOR_DELETION, VALID_ID_MARK_FOR_DELETION,
```

Comment:
From the tests it seems that the actual service, not  a  validator, is doing the validation. I think it should be a validator and the service tests would simply mock it. Or is this by design?

Reply:
Yes there no validator for the this service. Everything is done as part  of service. But in this case everything is mocked..no data base.

Reply:
I see there should be a validator but since this is a legacy service it is out of scope for your changes.